prescription/prescription.go

```diff
@@ -0,0 +1,127 @@
+	LoopMode          *string          `json:"loopMode,omitempty" bson:"loopMode,omitempty"`
+	AcknowledgedTerms *bool            `json:"acknowledgedTerms,omitempty" bson:"acknowledgedTerms,omitempty"`
+	State             *string          `json:"state,omitempty" bson:"state,omitempty"`
+	CreatedTime       *string          `json:"createdTime,omitempty" bson:"createdTime,omitempty"`
+	CreatedUserID     *string          `json:"createdUserId,omitempty" bson:"createdUserId,omitempty"`
+	ModifiedTime      *string          `json:"modifiedTime,omitempty" bson:"modifiedTime,omitempty"`
+	ModifiedUserID    *string          `json:"modifiedUserId,omitempty" bson:"modifiedUserId,omitempty"`
+	DeletedTime       *string          `json:"deletedTime,omitempty" bson:"deletedTime,omitempty"`
+	DeletedUserID     *string          `json:"deletedUserId,omitempty" bson:"deletedUserId,omitempty"`
+	Revision          *int             `json:"revision,omitempty" bson:"revision"`
+}
+
+type Weight struct {
+	Value *float64 `json:"value,omitempty" bson:"value,omitempty"`
+	Units *string  `json:"units,omitempty" bson:"units,omitempty"`
+}
+
+type Address struct {
+	Line1      *string `json:"line1,omitempty" bson:"line1,omitempty"`
+	Line2      *string `json:"line2,omitempty" bson:"line2,omitempty"`
+	City       *string `json:"city,omitempty" bson:"city,omitempty"`
+	State      *string `json:"state,omitempty" bson:"state,omitempty"`
+	PostalCode *string `json:"postalCode,omitempty" bson:"postalCode,omitempty"`
+}
+
+type InitialSettings struct {
+	BasalRateSchedule          *pump.BasalRateStartArray          `json:"basalSchedule,omitempty" bson:"basalSchedule,omitempty"`
+	BloodGlucoseTargetSchedule *pump.BloodGlucoseTargetStartArray `json:"bgTarget,omitempty" bson:"bgTarget,omitempty"`
+	CarbohydrateRatioSchedule  *pump.CarbohydrateRatioStartArray  `json:"carbRatio,omitempty" bson:"carbRatio,omitempty"`
+	InsulinSensitivitySchedule *pump.InsulinSensitivityStartArray `json:"insulinSensitivity,omitempty" bson:"insulinSensitivity,omitempty"`
+	BasalRateMaximum           *pump.BasalRateMaximum             `json:"basalRateMaximum,omitempty" bson:"basalRateMaximum,omitempty"`
+	BolusAmountMaximum         *pump.BolusAmountMaximum           `json:"bolusAmountMaximum,omitempty" bson:"bolusAmountMaximum,omitempty"`
+	PumpType                   *PumpType                          `json:"pumpType" bson:"pumpType"`
+	CGMType                    *CGMType                           `json:"cgmType" bson:"cgmType"`
+	// TODO: Add Suspend threshold - Dependent on latest data model changes
+	// TODO: Add Insulin model - Dependent on latest data model changes
+	// TODO: Add Bolus schedule? Does not exist in current pump settings model.
+}
+
+type PumpType struct {
+	Manufacturers *[]string `json:"manufacturers,omitempty" bson:"manufacturers,omitempty"`
+	Model         *string   `json:"model,omitempty" bson:"model,omitempty"`
```

Comment:
We may need a way to share the "master" list of valid/supported manufacturers & models with the client apps, to avoid replicating the list everywhere. I expect that the Web UX for creating prescriptions will have to provide some way to select the device(s) from a list, rather than a free-form entry. Ultimately, the Loop app can only accept the device(s) that it supports.

Reply:
Does the client go to the backend to get information?  My understanding is that the client (Loop) just sends info to the backend.  I always assumed that the client has the option of not using a backend

Reply:
Can the prescriber change the prescription after the initial one?  Do we keep both "initialSettings" objects around?  

I do think that the settings object is fine for recording the initial prescription.

Reply:
The user will have the ability to retrieve a prescription (with the initial settings) from the Loop app, thus fetching info from the backend. They can be changed but we will keep the full revision history.